### PR TITLE
Update delete command to incorporate all flag and varargs

### PR DESCRIPTION
**Pull request by cjyothika**
Delete command new proposed format: "delete all" - clears all modules
"delete /m MA1521 /m MA2002..."

### CODE REVIEW

File: src/main/java/seedu/modtrek/logic/parser/DeleteCommandParser.java [hunk 1]
```diff
@@ -17,9 +22,22 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
...
     public DeleteCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize(args, PREFIX_CODE);
         try {
-            Code code = ParserUtil.parseCode(args);
-            return new DeleteCommand(code);
+            boolean isAll;
+            Set<Code> codes;
+            String flag = argumentMultimap.getPreamble();
+            if (flag.equals("all")) {
```
Comment (Jun-How) [inline src/main/java/seedu/modtrek/logic/parser/DeleteCommandParser.java:31]: I think there needs to add a check for any arguments after delete all.

```suggestion
            if (flag.equals("all") && argumentMultimap.getValue(PREFIX_CODE).isEmpty()) {
```

**Review by Jun-How — COMMENTED** [review]
Generally I think it looks okay, with some improvements below. I also stated a possible suggestion below to improve on the functionality?

File: src/main/java/seedu/modtrek/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -3,55 +3,92 @@
...
+                Module moduleToDelete = new Module(code);
+                model.deleteModule(moduleToDelete);
+            } catch (ModuleNotFoundException e) {
+                throw new CommandException(String.format(MESSAGE_DELETE_MODULE_NOT_FOUND, code));
+            }
         }
-        moduleToDelete = lastShownList.get(index);
 
-        model.deleteModule(moduleToDelete);
         model.updateFilteredModuleList(PREDICATE_SHOW_NO_MODULES);
```
[the remark this answers is not in the document]
Comment (cjyothika) [reply src/main/java/seedu/modtrek/logic/commands/DeleteCommand.java:84]: I think it's show no modules according to the tests/previous AB3 implementation but you can change it if you want I guess.

File: src/main/java/seedu/modtrek/logic/commands/DeleteCommand.java [hunk 3]
```diff
@@ -3,55 +3,92 @@
...
-        if (index < 0) {
-            throw new CommandException(String.format(MESSAGE_DELETE_MODULE_NOT_FOUND, targetCode));
+
+        for (Code code : targetCodes) {
+            try {
+                Module moduleToDelete = new Module(code);
+                model.deleteModule(moduleToDelete);
+            } catch (ModuleNotFoundException e) {
+                throw new CommandException(String.format(MESSAGE_DELETE_MODULE_NOT_FOUND, code));
+            }
```
[the remark this answers is not in the document]
Comment (cjyothika) [reply src/main/java/seedu/modtrek/logic/commands/DeleteCommand.java:81]: ok that works too I guess. sounds reasonable.

**Comment by cjyothika** [comment]
> Generally I think it looks okay, with a few possible improvements below. I also stated a possible suggestion below to improve on the functionality? I can make the proposed changes if okay.

Yup, do you want to merge the changes before making a new PR?

**Comment by Jun-How** [comment]
Sure.

File: src/main/java/seedu/modtrek/logic/parser/DeleteCommandParser.java (hunk 1 above)
Comment (Jun-How) [reply src/main/java/seedu/modtrek/logic/parser/DeleteCommandParser.java:31]: oops nvm ignore this comment.

**Pull request merged**